packages/compiler-core/src/transforms/vOn.ts

```diff
@@ -28,6 +30,51 @@ export interface VOnDirectiveNode extends DirectiveNode {
   exp: SimpleExpressionNode | undefined
 }
 
+export const unfoldExpression = (
```

Comment:
There's really no need for this, compound expressions can be merged by simply merging their children.